pkg/controller/v1alpha2/experiment/experiment_controller.go

```diff
@@ -368,6 +368,26 @@ func (r *ReconcileExperiment) createTrials(instance *experimentsv1alpha2.Experim
 }
 
 func (r *ReconcileExperiment) deleteTrials(instance *experimentsv1alpha2.Experiment, deleteCount int32) error {
+	logger := log.WithValues("Experiment", types.NamespacedName{Name: instance.GetName(), Namespace: instance.GetNamespace()})
+	trials := &trialsv1alpha2.TrialList{}
+	labels := map[string]string{consts.LabelExperimentName: instance.Name}
+	lo := &client.ListOptions{}
+	lo.MatchingLabels(labels).InNamespace(instance.Namespace)
 
+	if err := r.List(context.TODO(), lo, trials); err != nil {
+		logger.Error(err, "Trial List error")
+		return err
+	}
+	actualDeleteCount := int(deleteCount)
+	if int32(len(trials.Items)) < deleteCount {
```

Comment:
when do this condition happen?

Reply:
No idea, just to make sure the program won't crash.

Reply:
I am not sure if it is needed. deleteCount is updated on the results of List operation. https://github.com/kubeflow/katib/blob/master/pkg/controller/v1alpha2/experiment/experiment_controller.go#L280 

One question is if List operation returns stale entries. But this is not solved here too

Reply:
Because we do not use the trials from https://github.com/kubeflow/katib/blob/master/pkg/controller/v1alpha2/experiment/experiment_controller.go#L280 . Thus the result may be not consistent, IMO.

Reply:
This just to avoid IndexOutOfBoundsException

Reply:
What if you use the earlier `trial.Items` results  than a new List operation?

Reply:
It works for me. I am working on it now.